spec/models/subscription_spec.rb

```diff
@@ -61,6 +54,32 @@
     end
   end
 
+  context 'reference' do
+    context 'when common search criteria is provided' do
+      it 'generates a reference on initialization' do
+        subscription = Subscription.new(search_criteria: {
+          location: 'Somewhere', radius: 30, keyword: 'some keywords'
+        }.to_json)
+
+        expect(subscription.reference).to eq('Some keywords jobs within 30 miles of Somewhere')
+      end
+    end
+
+    context 'when no common search criteria is provided' do
+      it 'does not set a default reference' do
+        subscription = Subscription.new(search_criteria: { radius: 20, phase: 'primary' }.to_json)
```

Comment:
Could we possibly add support for phases, or some other filters too? Would be nice if we could catch the vast majority of cases

Reply:
I think we cover most with the two we're checking, and it makes the references a lot more complex to generate if we turn some on and off depending on what's there. If we don't it gets very long. Not sure what the best approach is.